### ISSUE TITLE
update hover and focus styles 

### ISSUE THREAD

**Issue by rlskoeser** [Princeton-CDH/ppa-django#360]
See #344 for the revised designs

improves accessibility (visible focus for keyboard navigation) and consistency

**Comment by gissoo** [comment]
Also, this is showing up when focusing on FF, Chrome, and Safari, just fyi: 
![Screen Shot 2021-05-20 at 12 44 56 PM](https://user-images.githubusercontent.com/44615389/119017613-31044480-b969-11eb-869e-c290379e5bfb.png)

**Comment by gissoo** [comment]
@thatbudakguy thanks for reminding me of #344!! and for the explanation on the submit query. 
- the focus style is now fixed 
- the hover style on the active buttons is also fixed
- thanks for fixing the underlines

- one more point: the hover style is also applied as the select style for the main nav (I mean the second when you select a menu item) but this is not showing up on Safari.

**Comment by thatbudakguy** [comment]
> the hover style is also applied as the select style for the main nav

so, this led me back to something we've talked about before: in most browsers (safari might be an exception; mobile safari in particular is generally garbage and never behaves properly) when you click something, you activate both the `:active` and `:focus` states at once. the reason for this is that `:focus` indicates when a control is the target of an interaction, so if you are currently "activating"/interacting with something, you are (by definition) also "focused on"/targeting it. this is the reason why clicking something will sometimes also show the focus indicator, which sticks around until you click/activate something else.

the newest solution to avoid this behavior for mouse users is something called `:focus-visible`, which basically just says "only apply the focus design when the object is focused by a keyboard, not when it's activated by a mouse". last time i tried to use it, it didn't appear to work, but i tried again and it now works!! so that should fix the issue of the main nav applying the hover/focus style when you click, and it also fixes the same problem with the page navigation buttons. let me know if this isn't the case for you (I tested on FF).

_however_, this strategy doesn't work for the collection buttons on the search. the reason for that is that while they look like buttons, i actually implemented them as styled checkboxes, because they need to be part of the search form and behave like checkboxes for screen readers (and also if you didn't load any CSS, they'd still work). unfortunately, that means `:focus-visible` doesn't work on them, because when you focus them, you're actually focusing an invisible checkbox! the thing that is visible is the checkbox's `<label>`, which i've styled to look like a button. in the future i might be able to find a way around this, but it wasn't straightforward to do here.

sorry for long-winded explanation; maybe it will prove useful for later reference.

**Comment by gissoo** [comment]
FYI, @thatbudakguy this is happening till I do hard refresh – fwtw: ![Screen Shot 2021-05-21 at 11 29 04 AM](https://user-images.githubusercontent.com/44615389/119161863-c2d28700-ba27-11eb-8d4e-a3bf76d42de2.png)

**Comment by thatbudakguy** [comment]
that sometimes happens after a deploy. if it went away after the refresh and you're happy with everything else, go ahead and close this

**Comment by gissoo** [comment]
@thatbudakguy thank you for explaining!! Super helpful, based on what you have written this means that Safari is doing it right for once :D – FYI, I'm testing these on desktop mostly since the hover and focus are not applicable there. 

A couple of points: 
- Nick, I don't see the focus style (the dashed outline) on the collections buttons though when I select them with my cursor, this contradicts what you said above. 
- The focus style for pagination and back to top button does not appear anymore on FF, Chrome, and Safari! :( it's still working on the collections buttons though.

**Comment by gissoo** [comment]
@thatbudakguy you said on Slack that you can see the focus style on pagination and back to top buttons fine – I checked and my chrome and safari are not the latest versions and am having difficulty updating them, but it’s weird that they were showing the focus styles yesterday, and are only not showing them for pagination and back to top buttons today. If you can see them on Chrome and Safari that’s great, I’ll go ahead and close the issue.
